fiftyone/core/models.py

```diff
@@ -1097,7 +1090,11 @@ def compute_embeddings(
 def _compute_image_embeddings_single(
     samples, model, embeddings_field, skip_failures, progress
 ):
-    samples = samples.select_fields()
+    if embeddings_field is not None and "." in embeddings_field:
+        root_field = embeddings_field.split(".", 1)[0]
+        samples = samples.select_fields(root_field)
+    else:
+        samples = samples.select_fields()
```

Comment:
These 6 added blocks are all the same. Can we move them to a function?

Reply:
I was following the existing pattern of duplicating the logic in each of the `_compute_XXX_for_YYY()` methods, but, sure, just pushed https://github.com/voxel51/fiftyone/pull/6374/commits/7a9dba3bb5e63065ad755981353d2bfe78db68e7 that refactors all of these into shared utils.